packages/@atjson/document/src/query.ts

```diff
@@ -1,8 +1,40 @@
+import Document from '.';
 import Annotation from './annotation';
-import Document from './index';
 
-export interface Filter {
-  [key: string]: any;
+export type DeprecatedTransform = (annotation: JoinableAnnotation) => Annotation | null;
+export type Transform = (annotation: JoinableAnnotation) => TransformResult;
+
+export interface TransformResult {
+  add?: JoinableAnnotation[];
+  remove?: JoinableAnnotation[];
+  retain?: JoinableAnnotation[];
```

Comment:
Is `retain` needed; also, could you add a comment for this, since it maaaay be a tad confusing for folks not aware of OT terminology.